sensor/common/sensor/sensor.go

```diff
@@ -410,10 +410,11 @@ func (s *Sensor) communicationWithCentralWithRetries(centralReachable *concurren
 		// At this point, we know that connection factory reported that connection is up.
 		// Try to create a central communication component. This component will fail (Stopped() signal) if the connection
 		// suddenly broke.
-		s.centralCommunication = NewCentralCommunication(s.reconnect.Load(), s.reconcile.Load(), s.components...)
+		centralCommunication := NewCentralCommunication(s.reconnect.Load(), s.reconcile.Load(), s.components...)
 		syncDone := concurrency.NewSignal()
-		s.centralCommunication.Start(central.NewSensorServiceClient(s.centralConnection), centralReachable, &syncDone, s.configHandler, s.detector)
-		go s.notifySyncDone(&syncDone)
+		centralCommunication.Start(central.NewSensorServiceClient(s.centralConnection), centralReachable, &syncDone, s.configHandler, s.detector)
+		go s.notifySyncDone(&syncDone, centralCommunication)
+		s.centralCommunication = centralCommunication
```

Comment:
Wouldn't that be safer?

```suggestion
		s.centralCommunication = centralCommunication
		go s.notifySyncDone(&syncDone, centralCommunication)
```

Reply:
Good catch! I moved it one line above (before calling `Start) since `Start` spawns a goroutine.